### PR TITLE
docs: nested sorting

### PR DESCRIPTION
**Pull request by saihaj**
documenting https://github.com/graphprotocol/graph-node/pull/3096

### CODE REVIEW

File: website/pages/en/querying/graphql-api.mdx [hunk 1]
```diff
@@ -49,6 +49,23 @@ When querying a collection, the `orderBy` parameter may be used to sort by a spe
...
+{
+  tokens(orderBy: owner__name, orderDirection: asc) {
+    id
+    owner {
+      name
+    }
+  }
+}
+```
+
```
Comment (azf20) [inline website/pages/en/querying/graphql-api.mdx:68]: Is it worth adding a note on constraints (e.g. only works one level deep, doesn't work for interfaces - anything else?)

Comment (azf20) [reply website/pages/en/querying/graphql-api.mdx:68]: presumably it doesn't work if the nested entities are arrays (rather than single linked entities)?

Comment (azf20) [reply website/pages/en/querying/graphql-api.mdx:68]: Presumably it _does_ work on derived entities? I think worth adding these kind of clarifiers

File: website/pages/en/querying/graphql-api.mdx [hunk 2]
```diff
@@ -49,6 +49,23 @@ When querying a collection, the `orderBy` parameter may be used to sort by a spe
 }
 ```
 
+#### Example for nested entity sorting
+
+Sorting on the basis of nested entities is useful if you are looking to sort entities based on child fields.
```
Comment (azf20) [inline website/pages/en/querying/graphql-api.mdx:54]: This is a bit of a circular sentence - I might make it more of a statement of fact:
"As of Graph Node version 0.30.0, entities can be sorted by fields on nested entities. In the following example..."

**Review by azf20 — CHANGES_REQUESTED** [review]
Nice! Requesting a couple more clarifications and I think will be good to go.

File: website/pages/en/querying/graphql-api.mdx [hunk 3]
```diff
@@ -49,6 +49,25 @@ When querying a collection, the `orderBy` parameter may be used to sort by a spe
...
+  tokens(orderBy: owner__name, orderDirection: asc) {
+    id
+    owner {
+      name
+    }
+  }
+}
+```
+
+> Currently you can sort by one-level deep `String` or `ID` types on `@entity` and `@derivedFrom` fields. Unfortunately, [sorting by interfaces on one level-deep entities](https://github.com/graphprotocol/graph-node/pull/4058) and sorting by nested entities is not supported yet.
```
Comment (azf20) [inline website/pages/en/querying/graphql-api.mdx:69]: > sorting by nested entities is not supported yet
Do you mean arrays?

**Review by azf20 — CHANGES_REQUESTED** [review]
LGTM - one question on the last sentence

File: website/pages/en/querying/graphql-api.mdx (hunk 3 above)
Comment (saihaj) [reply website/pages/en/querying/graphql-api.mdx:69]: both 
CleanShot 2023-03-09 at 15 55 42

Comment (azf20) [reply website/pages/en/querying/graphql-api.mdx:69]: I see what you mean - I think worth being very specific then, i.e. "sorting by fields which are arrays, or which themselves are nested entities is not supported". I wonder why we can't use the ID of the `AnotherEntity` to sort, as that will just be a string (except for derived fields)?

Comment (saihaj) [reply website/pages/en/querying/graphql-api.mdx:69]: > I wonder why we can't use the ID of the AnotherEntity to sort, as that will just be a string (except for derived fields)?

I think then it makes it go deeper in the tree. Technically it is just recursively going down the selection set. But we are limiting to 1 level cause of performance reasons.

**Review by dotansimha — APPROVED** [review]

**Review by azf20 — APPROVED** [review]

**Pull request merged**